### PR TITLE
`Development`: Add screencasts for build agent and queue features

### PR DESCRIPTION
**Pull request by laurenzfb**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

The documentation for ICL is incomplete as screencasts for the newly added features are missing.

### Description

I added screencasts for the agent and queue features. I also moved the ICL setup up before Jenkins and Gitlab, as it is the prefered setup.

### Review Progress

https://artemis-platform--8737.org.readthedocs.build/en/8737/user/exercises/programming.html#build-agent-view
https://artemis-platform--8737.org.readthedocs.build/en/8737/user/exercises/programming.html#build-queue-view
https://artemis-platform--8737.org.readthedocs.build/en/8737/dev/setup.html

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

**Review by rstief — APPROVED** [review]
documentation looks good

File: docs/dev/setup/integrated-code-lifecycle.rst [hunk 1]
```diff
@@ -3,10 +3,9 @@
 Integrated Code Lifecycle Setup
 -------------------------------
 
-This section describes how to set up a programming exercise environment based on Integrated Code Lifecycle.
+This section describes how to set up a programming exercise environment based on Integrated Code Lifecycle which includes a local Version Control system and a local Continuous Integration system.
```
Comment (florian-glombik) [inline docs/dev/setup/integrated-code-lifecycle.rst:6]:
```suggestion
This section describes how to set up a programming exercise environment based on the Integrated Code Lifecycle, which includes a local Version Control system and a local Continuous Integration system.
```

File: docs/dev/setup/integrated-code-lifecycle.rst [hunk 2]
```diff
@@ -3,10 +3,9 @@
...
 These two systems are integrated into the Artemis server application and thus the setup is greatly simplified compared to the external options.
```
Comment (florian-glombik) [inline docs/dev/setup/integrated-code-lifecycle.rst:7]: Not directly related, but we could address it aswell

```suggestion
These two systems are integrated into the Artemis server application, and thus, the setup is greatly simplified compared to the external options.
```

**Review by florian-glombik — CHANGES_REQUESTED** [review]
Looks good except for the typos

The second video had a rather bad resolution, still acceptable, but if that only happened at the compression we might want to upload a video with a higher resolution there 😅

**Review by BBesrour — DISMISSED** [review]
Looks good!
Small nitpick, as Florian pointed out, the quality of the second screencast is a slightly worse than the first one

**Comment by laurenzfb** [comment]
@florian-glombik @BBesrour Thanks for the suggestions! I'll try to get a better version of the second screencast!

**Review by rstief — APPROVED** [review]

**Review by BBesrour — APPROVED** [review]
Thanks for improving the quality! Looks good!

**Review by florian-glombik — APPROVED** [review]
Thank you for fixing the typos and uploading the video with higher resolution!

**Review by egekurt123 — APPROVED** [review]
Screencasts LGTM

**Pull request merged**